app/main/views/reset_password.py

```diff
@@ -150,88 +150,63 @@ def update_password(token):
 @main.route('/change-password', methods=["GET", "POST"])
 @login_required
 def change_password():
-    form = PasswordChangeForm()
+    form = PasswordChangeForm(data_api_client=data_api_client, current_app=current_app, current_user=current_user)
     dashboard_url = get_user_dashboard_url(current_user)
 
-    if request.method == 'POST':
-        if form.validate_on_submit():
-            # Make sure old password is valid
-            user_json = data_api_client.authenticate_user(
-                current_user.email_address,
-                form.old_password.data)
-            if user_json:
-                response = data_api_client.update_user_password(
-                    current_user.id,
-                    form.password.data,
-                    updater=current_user.email_address
+    if form.validate_on_submit():
```

Comment:
Looks good, however I'd add a comment here to mention that the `old_password` check is now happening during `validate_on_submit` - I can foresee a dev looking at this code in the future (i.e. forgetful me, in 6 months' time) and deciding to be 'helpful' by adding it back in...

Reply:
Done 👍